gubernator/timestamp_parser.py

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env python
+# Copyright 2016 The Kubernetes Authors All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import regex
+
+def sort_by_timestamp(lines):
+    # Given an array of log lines, returns an array of the lines sorted by timestamp
+    timestamp_dict = {}
```

Comment:
This should be something closer to `return sorted(lines, key=regex.extract_timestamp)` -- don't build the intermediate structure like this. And if it's that simple, it doesn't need its own file.